kernel/src/vm/vmar/mod.rs

```diff
@@ -872,6 +872,67 @@ impl VmarInner {
         return_errno_with_message!(Errno::ENOMEM, "Cannot find free region for mapping");
     }
 
+    /// Allocates a free region for mapping, searching from high address to low address.
+    ///
+    /// If no such region is found, return an error.
+    fn alloc_free_region_topdown(&mut self, size: usize, align: usize) -> Result<Range<Vaddr>> {
+        // This value represents the highest possible address for a new mapping.
+        // For simplicity, we use a fixed value `512` here. The value contains the following considerations:
+        // - The stack fixed padding size.
+        // - The stack random padding size.
+        // - The future growth of the stack.
+        // FIXME: This value should consider the process's actual stack configuration, which may
+        // exist in ResourceLimits.
+        let high_limit = VMAR_CAP_ADDR - INIT_STACK_SIZE - PAGE_SIZE * 512;
+
+        let low_limit = VMAR_LOWEST_ADDR;
+        let try_alloc_in_hole_topdown = |hole_start: Vaddr,
```

Comment:
It is possible to define a method here (a `fn` item can exist in another `fn`):
```rust
fn try_alloc_in_hole_topdown(...) { ... }
```

This does not need to be a closure because it does not capture anything.